File: JS/solution_pg_신규아이디추천.js
function solution(new_id) {
  let answer = new_id.toLowerCase();
  answer = answer.replace(/[^a-z0-9-_.]/g, '');
  answer = answer.replace(/\.{2,}/g, '.');
  answer = answer.replace(/^\./, '');
  answer = answer.replace(/\.$/, '');
  if (answer === '') answer = 'a';
  if (answer.length > 15) answer = answer.substring(0, 15);
  answer = answer.replace(/\.$/, '');
  let a = answer.substring(answer.length - 1);
  while (answer.length < 3) answer += a;
  return answer;
}

File: Java/~2022/프로그래머스/Solution_pg_신규아이디추천.java
package programmers;

public class Solution_pg_신규아이디추천 {

	public static void main(String[] args) {
		System.out.println(solution("abcdefghijklmn.p"));
	}
	public static String solution(String new_id) {
        String answer = new_id.toLowerCase();
        answer = answer.replaceAll("[^\\w-.]*", "");
        answer = answer.replaceAll("\\.{2,}", ".");
        answer = answer.replaceAll("^[.]|[.]$","");
        if(answer.length() == 0)
        	answer = "a";
        if(answer.length() > 15) {
        	answer = answer.substring(0, 15);
        	answer = answer.replaceAll("[.]$", "");
        }
        while(answer.length() < 3)
        	answer = answer.concat(answer.substring(answer.length()-1));
        return answer;
    }

}
